src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -3659,8 +3659,12 @@ - (void)invokeCommands:(NSArray<NSArray<MTRCommandWithRequiredResponse *> *> *)c
     for (NSArray<MTRCommandWithRequiredResponse *> * commandGroup in [commands reverseObjectEnumerator]) {
         // We want to invoke all the commands in the group in order, propagating along the list of
         // current responses.  Build up that linked list of command invokes via chaining the completions.
+        mtr_weakify(self);
         for (MTRCommandWithRequiredResponse * command in [commandGroup reverseObjectEnumerator]) {
             auto commandInvokeBlock = ^(BOOL allSucceededSoFar, NSArray<MTRDeviceResponseValueDictionary> * previousResponses) {
+                mtr_strongify(self);
+                VerifyOrReturn(self, MTR_LOG_DEBUG("invokeCommands commandInvokeBlock called back with nil MTRDevice"));
```

Comment:
Is the right thing to just return and never call the original completion?  Or to synthesize an error for the command we are not invoking and call the next completion?

Reply:
Originally I thought we should just return. But you make a good point. I'll have it call next completion with error.

Reply:
Actually, I just reverted the change to call next completion. Rethinking this: any client that wishes to get completion callbacks should hold onto a reference of the MTRDevice object, IMO.

Reply:
Agree.

Reply:
Works for me.  I just wanted to make sure we are doing the expectations about this right.